### PR TITLE
Opal 299

### PR DESCRIPTION
**Pull request by AB-USAF**

### CODE REVIEW

File: weave/__init__.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 from .pantry import Pantry
 from .metadata_db import load_mongo
 
-__version__ = "0.14.7"
+__version__ = "1.0.0"
```
Comment (JA-USAF) [inline weave/__init__.py:28]: Whoa 😎

File: weave/tests/test_index_sqlite.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
+"""Pytest tests for the index directory."""
```
Comment (JA-USAF) [inline weave/tests/test_index_sqlite.py:1]: Maybe specify sqlite index tests.

File: weave/tests/test_index_sqlite.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        pantry_path=pantry_1_path,
+        file_system=test_pantry.file_system,
+        sync=True,
+    )
+
+    pantry_2 = Pantry(
+        IndexSQLite,
+        pantry_path=pantry_2_path,
+        file_system=test_pantry.file_system,
+        sync=True,
```
Comment (JA-USAF) [inline weave/tests/test_index_sqlite.py:80]: I believe sync=True is only used in IndexPandas. Not required for sqlite, remove here and elsewhere.

Comment (AB-USAF) [reply weave/tests/test_index_sqlite.py]: I believe I removed them all

File: weave/tests/test_index_sqlite.py (hunk 2 above)
Comment (AB-USAF) [reply weave/tests/test_index_sqlite.py]: yes of course

**Review by JA-USAF — APPROVED** [review]

File: weave/tests/test_index_sqlite.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    test_pantry.file_system.mkdir(pantry_1_path)
+    test_pantry.file_system.mkdir(pantry_2_path)
+
+    tmp_txt_file = test_pantry.tmpdir.join("test.txt")
+    tmp_txt_file.write("this is a test")
+
+    upload_path_1 = os.path.join(pantry_1_path, "text.txt")
+    upload_path_2 = os.path.join(pantry_2_path, "text.txt")
+
+    test_pantry.file_system.upload(str(tmp_txt_file.realpath()), upload_path_1)
```
Comment (gzusaf) [inline weave/tests/test_index_sqlite.py:65]: why is this upload required?

File: weave/tests/test_index_sqlite.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    # Remove the .db files.
+    os.remove(pantry_1.index.db_path)
+    os.remove(pantry_2.index.db_path)
+
+
+def test_index_pantry_with_same_name(test_pantry):
+    """Validate that 2 pantries with the same name will have unique indices
+    """
+    pantry_1_path = os.path.join(test_pantry.pantry_path, "test-pantry-1")
+    pantry_2_path = os.path.join(test_pantry.pantry_path,
```
Comment (gzusaf) [inline weave/tests/test_index_sqlite.py:128]: This is another section where the distinction between pantry_name and pantry_path is confusing. From what I understand the pantry_name is the pantry_path, but the paths of these 2 pantries are different.
.pantry_path/test-pantry-1 is not the same as .pantry_path/test/test-pantry-1

**Review by JA-USAF — CHANGES_REQUESTED** [review]
Ya I have the same question about the fs.uploads

File: weave/tests/test_index_sqlite.py [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+    assert pantry_index_2["weave_version"][0] == pantry2_basket.weave_version
+    assert pantry_index_2["address"][0] == pantry2_basket.address
+    assert pantry_index_2["storage_type"][0] == pantry2_basket.storage_type
+
+    # Remove the .db files.
+    os.remove(pantry_1.index.db_path)
+    os.remove(pantry_2.index.db_path)
+
+
+def test_index_pantry_with_same_name(test_pantry):
```
Comment (BE-USAF) [inline weave/tests/test_index_sqlite.py:124]: Why is this needed? If there is only one index in the pantry, why would we not just update that .db? Unless I'm missing something, I don't see why there should ever be more than one .db and don't know why the indexing would be necessary.

I also think if we're just testing the file name, why do we need to be uploading and checking all these files?

Let's talk about this.

**Review by BE-USAF — CHANGES_REQUESTED** [review]

File: weave/tests/test_index_sqlite.py [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+    assert pantry_index_1["address"][0] == pantry1_basket.address
+    assert pantry_index_1["storage_type"][0] == pantry1_basket.storage_type
+
+    assert pantry_index_2["uuid"][0] == pantry2_basket.uuid
+    assert isinstance(pantry_index_1["upload_time"][0], datetime)
+    assert pantry_index_2["parent_uuids"][0] == pantry2_basket.parent_uuids
+    assert pantry_index_2["basket_type"][0] == pantry2_basket.basket_type
+    assert pantry_index_2["weave_version"][0] == pantry2_basket.weave_version
+    assert pantry_index_2["address"][0] == pantry2_basket.address
+    assert pantry_index_2["storage_type"][0] == pantry2_basket.storage_type
```
Comment (CW-USAF) [inline weave/tests/test_index_sqlite.py:117]: I feel like we could condense these assert statements into just a few lines and test the same things

**Review by CW-USAF — CHANGES_REQUESTED** [review]

**Comment by AB-USAF** [comment]
> Ya I have the same question about the fs.uploads

File: weave/tests/test_index_sqlite.py (hunk 7 above)
Comment (gzusaf) [reply weave/tests/test_index_sqlite.py:117]: I feel like we could have more asserts actually. Are we certain that pantry_index_1["uuid"][0] != pantry2_basket.uuid if we're not asserting it

File: weave/tests/test_index_sqlite.py [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+        file_system=test_pantry.file_system,
+    )
+
+    pantry_2 = Pantry(
+        IndexSQLite,
+        pantry_path=pantry_2_path,
+        file_system=test_pantry.file_system,
+    )
+
+    assert pantry_1.index != pantry_2.index
```
Comment (BE-USAF) [inline weave/tests/test_index_sqlite.py:121]: Do we have an equals operator implemented? It may be beneficial to do pantry.index.to_pandas_df() then convert the DataFrames.

Comment (BE-USAF) [inline weave/tests/test_index_sqlite.py:121]: Do we have an equals operator implemented? What exactly are we checking for equality?

File: weave/tests/test_index_sqlite.py [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+    pantry_1_index = pantry_1.index.to_pandas_df()
+    pantry_2_index = pantry_2.index.to_pandas_df()
+
+    # Validate that each basket was uploaded to the proper pantry, and
+    # not the other.
+    assert not pantry_1_index.equals(pantry_2_index)
+    assert "0000" in pantry_1_index["uuid"].values
+    assert "0000" not in pantry_2_index["uuid"].values
+    assert "0001" in pantry_2_index["uuid"].values
+    assert "0001" not in pantry_1_index["uuid"].values
```
Comment (BE-USAF) [inline weave/tests/test_index_sqlite.py:182]: Love this

**Review by CW-USAF — APPROVED** [review]

**Review by JA-USAF — DISMISSED** [review]

**Review by JA-USAF — APPROVED** [review]

**Review by JA-USAF — APPROVED** [review]

**Review by BE-USAF — APPROVED** [review]
